modules/RTCStats/RTCStats.ts

```diff
@@ -65,6 +67,71 @@ class RTCStats {
         this._initialized = true;
     }
 
+    isTraceAvailable() {
+        return this._trace !== null;
+    }
+
+    /**
+      * A JitsiConnection instance is created before the conference is joined, so even though
+     * we don't have any conference specific data yet, we can initialize the trace module and
+     * send any logs that might of otherwise be missed in case an error occurs between the connection
+     * and conference initialization.
+     *
+     * @param connection - The JitsiConnection instance.
+     * @returns {void}
+     */
+    startWithConnection(connection: JitsiConnection) {
+        const { options } = connection;
+        const name = options?.name ?? '';
+        const {
+            analytics: {
+                rtcstatsUseLegacy: useLegacy = false,
+                rtcstatsEndpoint: endpoint = '',
+                rtcstatsEnabled = false
+            } = {},
+        } = options;
+
+        // Even though we have options being passed to init we need to recheck it as some client (react-native)
+        // don't always re-initialize the module and could create multiple connections with different options.
+        if (!rtcstatsEnabled) return;
```

Comment:
What if you open a deployment that has rtcStats enabled and you initialize trace. Then you move to another deployment which does not have rtcStats enabled, don't we need here to reset trace dispose something ... so we do not send logs and data to the initial connection place?
What if just the rtcStats connection url changes, so you connect first to stage then to prod ... 

Are the above cases covered?

Reply:
This should be covered, whenever we do traceConnect, which happens either when you initialize the JitsiConnection or you join a JitsiConference, we read the config options and reset the trace.